server.js

```diff
@@ -50,11 +51,23 @@ if (config.auth && config.auth.plugin && config.auth.enabled !== false) {
 
 // Install the Marklogic Provider after installing an authorization plugin.
 const provider = require('./src/koop');
+const https = require("https");
 koop.register(provider);
 
 // Create and configure the Express app server.
 const app = express();
-const port = config.port || 8080;
 app.use('/', koop.server);
-app.listen(port);
-log.info("Koop listening on port " + port + "; press ctrl-C to exit");
+
+let port = config.port || 8080;
+if (config.ssl.enabled) {
+  const https = require('https');
+  const options = {
+    key: fs.readFileSync(config.ssl.key, 'utf8'),
+    cert: fs.readFileSync(config.ssl.cert, 'utf8')
+  };
+  port = config.ssl.port || 443;
+  https.createServer(options, app).listen(port);
+} else {
+  app.listen(port);
+}
+log.info(`Koop MarkLogic Provider listening on ${config.ssl.port}`);
```

Comment:
Sorry one more tweak - this is always printing the SSL port, which isn't correct. Just do the previous log message - `Koop listening on ${port}; press ctrl-C to exit` .

Reply:
ugh, sorry